### PR TITLE
internal: Clean up path handling a little bit

### PR DESCRIPTION
**Pull request by tronical**
* Remove the undocumented and dysfunctional PathLayout
* Simplify the empty-path code path and avoid a property dependency in the common case

### CODE REVIEW

File: internal/core/items/path.rs [hunk 1]
```diff
@@ -148,7 +148,7 @@ impl Path {
         let viewbox_width = self.viewbox_width();
         let viewbox_height = self.viewbox_height();
 
-        let mut elements_iter = self.elements().iter();
+        let mut elements_iter = self.elements().iter()?;
```
Comment (ogoffart) [inline internal/core/items/path.rs:151]: Maybe move that up in the function so we don't query all the other property in case the path is empty

**Review by ogoffart — COMMENTED** [review]
Sad to see PathLayout just go...  But well, we can bring it back later indeed.

File: internal/core/items/path.rs (hunk 1 above)
Comment (tronical) [reply internal/core/items/path.rs:151]: Good idea

**Comment by tronical** [comment]
> Sad to see PathLayout just go... But well, we can bring it back later indeed.

Me, too, a little. It seems like a useful feature. As I started refactoring the Path code I noticed that I'd have to adjust the path layout code and then a red light started flashing ;-)

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -298,4 +299,9 @@ jobs:
                   --skip test_interpreter_crashes_issue1271_set_in_if \
                   --skip test_interpreter_models_assign_equal_model \
                   --skip example_carousel \
-                  --skip example_fancy_demo
+                  --skip example_fancy_demo \
+                  --skip test_interpreter_elements_path_fit \
+                  --skip test_interpreter_layout_path \
+                  --skip test_interpreter_layout_path_alignment \
+                  --skip test_interpreter_layout_path_for \
+                  --skip test_interpreter_layout_path_offset
```
Comment (ogoffart) [inline .github/workflows/ci.yaml:307]: I believe this works with substring, so --skip test_interpreter_layout_path_ should skip all the path layout ones

Comment (tronical) [reply .github/workflows/ci.yaml:307]: Ahh, ok, let's try that

**Review by ogoffart — APPROVED** [review]
(should probably be rebased)

**Pull request merged**